### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"

### DISCUSSION THREAD

**Discussion by Taraldinn** [community/community#11324] [Copilot Conversations] [answered]
![image](https://user-images.githubusercontent.com/59772487/153477359-b338b789-7455-45d4-8b29-71383fe7a7b8.png)
hey, I am just getting this error for few days ! I am trying to login in my copilot account but it says "GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"" .. can anyone help me ?

**Comment by Taraldinn** [comment]
no I am not ! I have just installed Vs Code Insider and I have successfully
installed copilot on that !

**Reply by Kong12384abc** [reply]
now u can solve this problem  ?

**Reply by PaNDanese** [reply]
I had the identical error message. 
Simply running the uninstaller for VS Code and then re-installing it didn't work (I got the same error message when trying to activate Copilot).

The only thing that worked for me was a **full** uninstall of VS Code (https://superuser.com/questions/1380208/how-to-completely-uninstall-visual-studio-code-from-windows-10) and then a reinstall.

After that, I was able to activate Copilot. 

Also, probably goes without saying, but you probably want to save a copy of your settings + list of extensions before you do the full uninstall.

**Reply by ajonno** [reply]
what a joke nice one MS

**Reply by oneplusiota** [reply]
@ajonno agree. I have been facing the issue for almost a month. I tried uninstalling the Copilot extension, but it doesn't seem to work. I think this is just another bug that MS needs to fix so that their services run for another month or so until they encounter this same issue again.

![Screenshot from 2024-01-14 23-16-30](https://github.com/community/community/assets/61914586/86fd873a-262b-4d16-90aa-4b996ffe9d2b)


This keeps happening every time I log in, even though I'm signed into VS code with my MS account.

**Comment by LCristinaM** [comment]
Hi, how did VS Code Insiders work for you, I had the same error and I managed to fix it by uninstalling VSC and downloading the Insiders version.

**Comment by fedevigo** [comment]
You probably have **GitHub Authentication disabled**.  
GitHub Authentication is an extension that is bundled with Visual Studio Code.  

Go to File -> Preferences -> Profiles -> Show Contents
![GitHub Copilot could not connect to server](https://user-images.githubusercontent.com/11222712/224421489-f00d1b0a-1b70-4624-b720-93582e4248a5.png)

In the left panel, in the "Extensions" group, click in "GitHub Authentication" and enable the extension.
![Extension activation failed](https://user-images.githubusercontent.com/11222712/224421572-61799d81-3b6a-4ae7-87d3-7a9c27824c45.png)

**Reply by LCristinaM** [reply]
Thank you so much!!! you saved my life!! 
I followed your instructions in my VSC and the problem was solved! 🥳   
The "GitHub Authentication" extension if it was disabled! 😮‍💨 
Now I have Github Copilot ready thanks to you! ❤️ 

![Screenshoot](https://user-images.githubusercontent.com/110795344/224428171-5c7d08fe-5aac-4b3d-a868-c68fffe895ac.png)

**Reply by Bit-Part-Young** [reply]
> You probably have **GitHub Authentication disabled**. GitHub Authentication is an extension that is bundled with Visual Studio Code.
> 
> Go to File -> Preferences -> Profiles -> Show Contents ![GitHub Copilot could not connect to server](https://user-images.githubusercontent.com/11222712/224421489-f00d1b0a-1b70-4624-b720-93582e4248a5.png)
> 
> In the left panel, in the "Extensions" group, click in "GitHub Authentication" and enable the extension. ![Extension activation failed](https://user-images.githubusercontent.com/11222712/224421572-61799d81-3b6a-4ae7-87d3-7a9c27824c45.png)

It works! Thanks!

**Reply by zol9527** [reply]
You save my ass. Thank you very much !!!!!

**Reply by DavidPr0** [reply]
Thank you very much, Copilot connected with success.

**Reply by dkozar** [reply]
This is it! Thank you! :) 

This hidden extension isn't visible in the "Extensions" view (since it's bundled with VSCode).

Without it ENABLED, one doesn't get the "Sign in with GitHub to use GitHub Copilot (1)" menu:

![2023-08-09_10-04-30](https://github.com/community/community/assets/1078403/a0134c3c-e4e3-45b0-91cc-d51ae14ab37b)

Also, if using the GitHub Copilot extension, you're getting the following error:

`GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"`

**Reply by rubenvl2** [reply]
that worked perfectly in ubuntu. You saved us! Thanks :)

**Reply by AnaiO** [reply]
Thank youuuuuuuuu <3

**Reply by niels-heinemann** [reply]
Jesses, Internet is full of "Unijnstall and purge code and all of your software" and you just hit the simple reason. Thanks a huge load!

**Reply by tit006** [reply]
Really appreciate, without the hint I can never figure it out.

**Reply by HenryChan-Git** [reply]
Thank you very much, it works!

**Reply by ViniNathan** [reply]
Guys, I found another solution. I can't find the GitHub Authentication in preferences>profiles>show content. So, I found an alternative: Open your VS Code, press CTRL+SHIFT+P, type Developer: Restart Extension Host, then press Enter

**Reply by duraanali** [reply]
Thanks, this worked for me, appreciate it!

**Reply by adrianocrippa** [reply]
Thank you!!

**Reply by antimeat** [reply]
you bloody legend thankyou!

**Reply by stevedekorte** [reply]
Thanks! This worked for me too. Sure would be nice if the VS Code auth error message mentioned that auth was disabled.

**Reply by Guilopesc1** [reply]
Thanks brooo! Worked for me too.

**Reply by kona3266** [reply]
Thanks bro! this worked for me.

**Reply by Chocyork** [reply]
hey bro! l cant find this😭

**Reply by sahil1330** [reply]
Thank you bro it really worked! you are really a legend.

**Reply by it-redbull** [reply]
thank you so much you saved me a lot of time

**Reply by wcy1014** [reply]
thanks bro，you are hero

**Reply by AzraelRex** [reply]
Heyo mate how do you save this setting? It keeps asking me to import a folder.

**Reply by HmEJ** [reply]
the best answer!

**Reply by hongla-tw** [reply]
Thank you bro.

**Reply by arnaudVLcst** [reply]
It works for me. Thx

**Reply by lwasser** [reply]
thank you!! this worked perfectly. that was the issue for me too!!

**Reply by JoeZhou2** [reply]
You save my whole day!

**Reply by cjauvin** [reply]
As of December 21 2024, on a Mac, just after the announcement of the new free Copilot offer, this is the only thing that worked for me, after a long search and of lot of frustration, many thanks!

**Reply by caldiver07** [reply]
Solved mine too.. THANKS.....

**Reply by SujeetAtwe789** [reply]
thanks its works for me ( thanks a lots )

**Reply by HFTrader** [reply]
Holy shit. I had been chasing this for hours! How did it get disabled though? I dont remember ever doing that!

**Comment by dkozar** [comment, marked as answer]
As @Cmolosa said, your GitHub Authentication extension is probably disabled (by default). Confirming it solved the problem for me.

**Comment by lumos42** [comment]
I'm on macOS, and getting the same error message. How do I fix?

**Reply by shorifulce** [reply]
<img width="766" alt="Screenshot 2023-04-08 at 23 52 38" src="https://user-images.githubusercontent.com/53480758/230744107-25a8d9de-dc94-472e-8a24-4c8849b65f8a.png">
Click on Show Contents, and enable GitHub globally. Then try to log in.
For me it worked.

**Comment by bellerofonte** [comment]
> You probably have **GitHub Authentication disabled**. GitHub Authentication is an extension that is bundled with Visual Studio Code.

@fedevigo any ideas how to enable **GitHub Authentication** inside a Remote-SSH connection? There is no such extension in the Profile tab and I cannot find one (`vscode.github-authentication`) in the marketplace.

**Reply by mateusfg7** [reply]
+1, I'm using Codespaces.

**Reply by longxiaotian799** [reply]
me too

**Reply by ViniNathan** [reply]
Have you guys already tried this?
Open your VS Code, press CTRL+SHIFT+P, type Developer: Restart Extension Host, then press Enter

**Reply by mateusfg7** [reply]
@ViniNathan yes, the error persists.

**Reply by longxiaotian799** [reply]
I suggest you can download vscode insiders edition not the original edition and configure it，because I cannnot find another solution

**Reply by mateusfg7** [reply]
I started codespaces again after a while and now it works. I don't know why, I didn't do anything.

**Comment by Yerock09** [comment]
Pues no pasa nada simplemente quería desde que ratos cambiar algo xke
sentía que me hackeaban y terminaste haciéndolo tu jaja

**Comment by skanderKhabou** [comment]
I had the same issue after viewing and reading comments , i figure out the problem , if in your preferrence you can't find the git hub auth extension it's normal , what you should do is sign out from github inside the vs code ( left accounts and click sign out ) after this you should reload vs code clicking on F1 and go to developer reload window , after this you should sign in again .. you can find those steps here : 
https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-issues-with-github-copilot-chat-in-ides

After this , you may have the same issue , what you should do is sync your settings in the same left account button , check also in the config button under the account if there is any thing to enable exemple profile setting etc .
No need to uninstall and reinstall vs code just sign out from github in vs code and follow the waves it will work . Good Luck

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by iam11332** [comment]
hi @Taraldinn , why I cannot find the Github Authentication in my vscode?
![image](https://github.com/community/community/assets/170870647/04a2e569-7216-497c-877f-4a1955d13536)

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！

**Comment by wcy1014** [comment]
邮件已收到，谢谢！
